### ISSUE TITLE
No history for unavailable services

### ISSUE THREAD

**Issue by Skleni** [Xabaril/AspNetCore.Diagnostics.HealthChecks#475]
**What happened**:

I have two projects: One I want to monitor (my app) and one for the Health Check UI (both ASP.NET Core 3.1). For testing, I added one random check to the app. It works as expected and the history can be seen on the dashboard:

![image](https://user-images.githubusercontent.com/10420514/78562361-84750b00-7819-11ea-8da1-a7c1a6ad0b77.png)

However, when I stop the app (so it can't even return that it's unhealthy anymore), this is not added to the history as "Unhealty". Instead, the whole history is gone:

![image](https://user-images.githubusercontent.com/10420514/78562544-da49b300-7819-11ea-8f60-a582039f7ef0.png)

**What you expected to happen**:

I would expect another "Unhealthy" entry to be added to the history, but the history is gone.

**How to reproduce it (as minimally and precisely as possible)**:

Create two projects as described above. Let the app run until history entries are available, the stop it.

**Source code sample**:

*App:*

```csharp
services.AddHealthChecks()
        .AddCheck("test", () => DateTime.Now.Second % 2 == 0 ? HealthCheckResult.Healthy() : HealthCheckResult.Unhealthy());

...

app.UseEndpoints(endpoints =>
{
    endpoints.MapHealthChecks("health", new HealthCheckOptions()
    {
        Predicate = _ => true,
        ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
    });
});
```
*Dashboard:*

```csharp
services.AddHealthChecksUI();

...

app.UseEndpoints(endpoints =>
{
    endpoints.MapHealthChecksUI();
});
```

**Environment**:

- .NET Core version: 3.1
- Healthchecks version: 3.0.11
- Operative system: Windows 10

**Comment by CarlosLanderas** [comment]
Hello @Skleni. Let me take a look into this issue. Thanks!

**Comment by Skleni** [comment]
Yes, it was already unhealthy. The two screenshots above were taken right before and after stopping the app. So I think it's correct that no new entry is added, but the existing history entries shouldn't be removed, should they?

**Comment by CarlosLanderas** [comment]
Yeah. We have just released a version with a long awaited feature. Database Providers. You can check version 3.1.1-preview1 with sqlserver sqlite, postgre and in-memory.
This new version does not remove the checks on startup. Except in memory ofc

**Comment by Skleni** [comment]
I got the same issue as reported by @cesarbmx in #455. I'll check again once that is resolved.

**Comment by CarlosLanderas** [comment]
@Skleni can you try again? ;) Thanks!

**Comment by Skleni** [comment]
Updating the package worked now, thanks a lot, I've been waiting for that change as well. :)

I still have my original issue though, but I think I now understand a bit better what causes it:

The history is available for the individual entries, but not for the service itself. When the service goes offline, it can't even return which entries should be there. So technically, there's no way for the HealthCheck UI to know the status of the individual entries...but I still think it would be better to add an "Unhealthy" history entry instead of resetting the whole history.

**Comment by CarlosLanderas** [comment]
Hello @Skleni, what do you mean by "resetting" the whole history?

**Comment by Skleni** [comment]
Let's say the service I'm monitoring has one check. Now the following things happen:

1) The check reports healthy. The service is displayed as healthy in the UI.
2) The check reports unhealthy. The service is displayed as unhealthy and the history shows two entries:
* Healthy
* Unhealthy
(see first screenshot above)
3) The service goes offline. The UI is unable to retrieve the checks response. The history now only shows "Unhealthy" (second screeshot above). The two entries that were previously there are gone.

**Comment by scabana** [comment]
We have the same issue. If the monitored service goes offline, it should most likely show the entries should be set to error or something like that. If the whole service goes offline, then it's entries should be considered offline, not just "gone".
